pybamm/parameters/parameter_values.py

```diff
@@ -954,3 +932,170 @@ def find_parameter(path):
+            for k in items.keys():
+                v = self[k]
+                if callable(v):
+                    # write the function body to the file
+                    function_output += inspect.getsource(v) + "\n"
+                    v = v.__name__
+                elif isinstance(v, tuple):
+                    # save the data to a separate csv file
+                    # and load it in the parameter set
+                    data_name = v[0]
+                    data_file_old = os.path.join(
+                        old_parameters_path,
+                        component_group.replace(" ", "_") + "s",
+                        self.chemistry[component_group],
+                        f"{data_name}.csv",
+                    )
+                    data_path = os.path.join(new_parameters_path, "data")
+                    if not os.path.exists(data_path):
+                        os.makedirs(data_path)
+                    data_file_new = os.path.join(data_path, f"{data_name}.csv")
+                    shutil.copyfile(data_file_old, data_file_new)
+
+                    # add data output
+                    if data_output == "":
+                        data_output = (
+                            "# Load data in the appropriate format\n"
+                            "path, _ = os.path.split(os.path.abspath(__file__))\n"
+                        )
+                    data_output += (
+                        f"{data_name} = pybamm.parameters.process_1D_data"
+                        f"('{data_name}.csv', path)\n"
+                    )
+                    v = f"pybamm.{data_name}"
+
+                    v = f"{data_name}"
+
+                # add line to the parameter output in the appropriate section
+                line_output = f'\n        "{k}": {v},'
+                if len(line_output) > 88:
+                    # this will be split into multiple lines by black
+                    line_output = f'\n        "{k}""": {v},'
+
+                dict_output += line_output
+
+        # save citation info
+        if hasattr(self, "citations"):
+            dict_output += (
+                "\n        # citations" + f"\n        'citations': {self.citations},"
+            )
+
+        # read README.md if they exist and save info
+        docstring = self._create_docstring_from_readmes(name)
+
+        # construct the output string
+        output = (
+            function_output
+            + data_output
+            + "\n# Call dict via a function to avoid errors when editing in place"
+            + "\ndef get_parameter_values():"
+            + docstring
+            + "\n    return {"
+            + dict_output
+            + "\n    }"
+        )
+
+        # Add more packages to preamble if needed
+        if "os." in output:
+            preamble += "import os\n"
+        output = preamble + "\n\n" + output
+
+        # Add pybamm. to functions that didn't have it in function body before
+        for funcname in [
```

Comment:
should we add a broader set of "standard" functions here or wait and add them until people need them?

Reply:
> @brosaplanella I kept the `testing_only` parameters and tests so that it still works with the old split structure in case people are still using that. I don't know if anyone is, nobody raised any complaints about this change. Do you think I should just remove it?

Makes sense, I agree we should keep it at least for a while.